keras_cv/layers/preprocessing/erasing.py

```diff
@@ -0,0 +1,226 @@
+        patch_value: Float. The value to fill in the patches. If None, will
+            patches with gaussian noise. Defaults to 0.0.
+    References:
+       [RandomErasing paper](https://arxiv.org/abs/1708.04896).
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    random_erase = keras_cv.layers.preprocessing.erasing.RandomErasing(1.0)
+    augmented_images, labels = random_erase(images, labels)
+    ```
+    """
+
+    def __init__(
+        self,
+        rate,
+        scale=(0.02, 0.33),
+        ratio=(0.3, 3.3),
+        patch_value=None,
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(rate, patch_value, seed, **kwargs)
+        self.scale = scale
+        self.ratio = ratio
+
+    def _compute_cut_size(self, batch_size, image_height, image_width):
+        area = tf.cast(image_height * image_width, tf.float32)
+        erase_area = area * tf.random.uniform(
+            [batch_size], minval=self.scale[0], maxval=self.scale[1]
+        )
+        aspect_ratio = tf.random.uniform(
+            [batch_size], minval=self.ratio[0], maxval=self.ratio[1]
+        )
+        h = tf.cast(tf.round(tf.sqrt(erase_area * aspect_ratio)), tf.int32)
+        w = tf.cast(tf.round(tf.sqrt(erase_area / aspect_ratio)), tf.int32)
+
+        h = tf.minimum(h, image_height - 1)
+        w = tf.minimum(w, image_width - 1)
+
+        return h, w
+
+    def get_config(self):
+        config = {
+            "scale": self.scale,
+            "ratio": self.ratio,
+        }
+        base_config = super(RandomErasing, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+class CutOut(BaseErasing):
+    """CutOut implements the CutOut data augmentation technique.
+
+    Args:
+        rate: Float between 0 and 1.  The fraction of samples to augment.
+        length: Integer. The side length of the square patches to cut out.
+        patch_value: Float. The value to fill in the patches. If None, will
+            patches with gaussian noise. Defaults to 0.0.
+    References:
+       [CutOut paper](https://arxiv.org/abs/1708.04552).
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    cutout = keras_cv.layers.preprocessing.erasing.CutOut(1.0, 50)
+    augmented_images, labels = cutout(images, labels)
+    ```
+    """
+
+    def __init__(self, rate, length, patch_value=0.0, seed=None, **kwargs):
+        super().__init__(rate, patch_value, seed, **kwargs)
+        self.length = length
+
+    def _compute_cut_size(self, batch_size, image_height, image_width):
+        length = tf.fill([batch_size], self.length)
+        return length, length
+
+    def get_config(self):
+        config = {"length": self.length}
+        base_config = super(CutOut, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+def _fill_rectangle(
```

Comment:
Let's refactor this to be in a shared `util.py` file and reuse it across `CutMix` and this.

Reply:
I have made a `keras_cv.utils.fill_utils` module containing a `fill_rectangle` function for now. This is not shared between `CutMix`, `CutOut`, and `RandomErasing`.

The decision to make a `keras_cv.utils` package was to mimic the utils structure in `keras`.